src/zenml/login/pro/utils.py

```diff
@@ -0,0 +1,121 @@
+            logger.debug(f"Failed to list tenants: {e}")
+        else:
+            if servers:
+                server = servers[0]
+                if server.status == TenantStatus.AVAILABLE:
+                    return (
+                        f"The '{server.name}' ZenML Pro server that the client "
+                        "is connected to is currently running but you may not "
+                        "have the necessary permissions to access it. Please "
+                        "contact your ZenML Pro administrator for more "
+                        "information or try to manage the server members "
+                        "yourself if you have the necessary permissions by "
+                        f"visiting the ZenML Pro tenant page at {server.dashboard_url}."
+                    )
+                if server.status == TenantStatus.DEACTIVATED:
+                    return (
+                        f"The '{server.name}' ZenML Pro server that the client "
+                        "is connected to has been deactivated. "
+                        "Please contact your ZenML Pro administrator for more "
+                        "information or to reactivate the server yourself if "
+                        "you have the necessary permissions by visiting the "
+                        f"ZenML Pro Organization page at {server.dashboard_organization_url}."
+                    )
+                if server.status == TenantStatus.PENDING:
+                    return (
+                        f"The '{server.name}' ZenML Pro server that the client "
+                        "is connected to is currently undergoing maintenance "
+                        "(e.g. being deployed, upgraded or re-activated). "
+                        "Please try again later or contact your ZenML Pro "
+                        "administrator for more information. You can also "
+                        f"visit the ZenML Pro tenant page at {server.dashboard_url}."
+                    )
+                return (
+                    f"The '{server.name}' ZenML Pro server that the client "
+                    "is connected to is currently in a failed "
+                    "state. Please contact your ZenML Pro administrator for "
+                    "more information or try to re-deploy the server "
+                    "yourself if you have the necessary permissions by "
+                    "visiting the ZenML Pro Organization page at "
+                    f"{server.dashboard_organization_url}."
+                )
+
+            return (
+                f"The ZenML Pro server at URL '{url}' that the client is "
+                "connected to does not exist or you may not have access to it. "
+                "Please check the URL and your permissions and try again or "
+                "connect your client to a different server by running `zenml "
+                "login` or by using a service account API key."
+            )
+
+    return (
```

Comment:
Shouldn't this tell people to login to zenml pro as the first cause of action, instead of contacting their admin?

Or do we only reach this place in case the user is trying to authenticate to a pro tenant with a service account?

Reply:
These hints are only printed as a very last resort. Authorization and connectivity errors are caught and handled separately.